en/guide/xml_schema.md

```diff
@@ -78,26 +84,37 @@ As a concrete example, the definition of the [LANDING_TARGET_TYPE](../messages/c
     </entry>
 ```
 
-The main `enum` tags/fields are:
-* `name`: The name of the enum (mandatory). This is a string of capitalized, underscore-separated words.
-* `description` (optional): A string describing the purpose of the enum
-* `entry` (optional): An entry (zero or more entries can be specified for each enum)
-* [deprecated](#deprecated) (optional): A tag indicating that the enum is deprecated.
-
 
 #### entry {#entry}
 
-The enum `entry` tags/fields are:
+The "normal" enum `entry` tags/fields are:
 * `name`: The name of the enum value (mandatory). This is a string of capitalized, underscore-separated words.
 * `value` (optional): The *value* for the entry (a number).
 * `description` (optional): A description of the entry.
-* [param](#param) (optional): A param value used in a [MAV_CMD](../messages/common.md#MAV_CMD) enum entry.
 * [deprecated](#deprecated) / [wip](#wip) (optional): A tag indicating that the enum is deprecated or "work in progress". 
 
+[MAV_CMD](../messages/common.md#MAV_CMD) entries may additionally define these tags/fields:
+* [param](#param) (optional): A param value.
+* `hasLocation` (optional): A boolean (default true) indicating the entry has lat/lon/alt location information in its param 5, 6, and 7 values respectively.
+* `isDestination` (optional): A boolean (default true) indicating that the entry contains location information (as per `hasLocation`) **and** that the vehicle should fly through the waypoint.
```

Comment:
@amilcarlucas @DonLakeFlyer Re

> `IsDestination` means that the vehicle will move to/through that point. Please read the PR comments it has been discussed in there.

The problem is that I didn't understand the PR answer either:
- Is the above definition correct?
- I assume that "fly through the specified location" means that the command corresponds to a position in which the command doesn't cause the vehicle to *stop* (unless -say- it is the last waypoint). 
   - So, for example, [MAV_CMD_NAV_WAYPOINT](https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_WAYPOINT)  `hasLocation` is true and `isDestination` is true. However for loiter commands isDestination is false. Correct?
   - What about for Takeoff? If there is no other commands then the vehicle will loiter, but if there is a waypoint next it will fly on - so is this isDestination = False or True?

Reply:
@DonLakeFlyer can you clarify the questions about `isDestination` ?

Reply:
Couple things:
* You shouldn't need to set both isDestination and hasLocation. They are mutually exclusive. If isDestination is true, then it always has loaction ifornation.
* A Loiter command has a location in it and the vehicle flies to that location and then loiters around it, hence isDestination is set for it. isDestination doesn't really mean the vehicle always "exactly flies through" the location. It means the ground station shoudl draw some sort of waypoint line to the location. If the ground station wants to get fancy and not draw a line through the center of the loiter but instead draw a line through the tangent to the loiter circle then thats up to the GCS.
* Since the vehicle flies to the Takeoff item (at least in PX4, but not in ArduPilot) it has isDestination=true.

You need to think about isDestination as a hint to the GCS for whether it needs to consider a waypoint line to the location somehow. Whereas hasLocation is a hint to the GCS that it needs to draw some sort of indicator at that location, but a waypoint line doesn't go through it.

I would change hasLocation to something like standaloneLocation to better indicate mutually exclusive us of the two bools.

There is also the concept of "altitude only" which means that the command only uses param7 as altitude and no lat/lon.

Reply:
@DonLakeFlyer thank you for the explanation. Naming is indeed difficult. So now I understand real examples of isDestination, can you give me one real example of where I would use `hasLocation` or `hasAltitude`?

I'm  not too worried about the mutual exclusiveness aspect of the naming, because we can fix this with tooling (only allow one in the schema). 

In any case, what about we make it clear they are display hints, in the same way that we do things like prinformat?: `displayLocation`, `displayAltitude`, `displayFlightPath`

If these are *really* mutually exclusive you could make this more obvious using "As": `displayAsLocation`, `displayAsAltitude`, `displayAsOnFlightPath`
(ie you're not saying that it is a location, you're saying it should be displayed as a location).

Reply:
> can you give me one real example of where I would use `hasLocation` or `hasAltitude`?

hasLocation - MAV_CMD_DO_SET_ROI_LOCATION
hasAlitutde - MAV_CMD_NAV_CONTINUE_AND_CHANGE_ALT

Reply:
> If these are _really_ mutually exclusive you could make this more obvious using "As": `displayAsLocation`, `displayAsAltitude`, `displayAsOnFlightPath`
> (ie you're not saying that it is a location, you're saying it should be displayed as a location).

That sort of works but I'm not sure it is less confusing.

Reply:
Thanks for real examples. I think I have a solution based on the field attribute  `display="bitmask"`. 

The attribute is `display`, the possible values are `location`, `altitude`, `flight_path`. Because there is just one attribute these are necessarily mutually exclusive. If you hate "flight_path" then we can use "destination" - I will just have to document what it means.

Reply:
display thing seems ok. I think destination is better than flight_path, since it goes with location better.

Reply:
@amilcarlucas If you agree with this, do you think you could change the schema to replace these two with an attribute `display` that can take values of `location` and `destination`?
If there is a description it should be 
* Location - "entry displayed as a standalone location - e.g. ROI"
* Destination - "entry displayed as destination on flight path"
```
<xs:attribute name="hasLocation" type="xs:boolean" default="true"/>    <!-- entry has lat/lon/alt location -->
<xs:attribute name="isDestination" type="xs:boolean" default="true"/>  <!-- entry is a destination -->
```
I'm going to tidy and merge this as it is now, and will update when/if you can make the above change.